boot/boot_serial/src/boot_serial.c

```diff
@@ -421,6 +421,34 @@ bs_upload(char *buf, int len)
     flash_area_close(fap);
 }
 
+static void
```

Comment:
This should be added as optional. Also worth to know that the protocol will respond to unknown command with error code - so is echo really needed?